### ISSUE TITLE
Add switch currency function

### ISSUE THREAD

**Issue by marksteward** [emfcamp/Website#390]
We never change prices for tickets, so it's safe to recalculate the total and reapply it to the payment.

**Comment by marksteward** [comment]
This is for non-bank-transfer payments

**Comment by marksteward** [comment]
This could be fixed by #152

**Comment by marksteward** [comment]
For now, I think it's enough just to identify the safe ("retryable") states for GoCardless and Stripe payments and add the same code as in #603. E.g.:  

  https://github.com/emfcamp/Website/commit/f64e921#diff-d109f415df9de44c241a81857e48614fR75.

**Comment by marksteward** [comment]
This is 5f9b029 in the admin pages, and seems to work. Should be easy to pull out to the purchases page.

**Comment by russss** [comment]
Yeah, I was going to add this but it's actually not clear how to display it clearly. Maybe just a "change currency" button on the /account/purchases screen but that's quite a lot of buttons.

**Comment by marksteward** [comment]
Looks like it was added with 68a9041
